cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -397,46 +398,48 @@ private Metadata getSnapshotBeforeTime(Id.NamespacedId targetId, long timeMillis
    */
   public List<MetadataEntry> search(final String namespaceId, final String searchQuery,
                                     final MetadataSearchTargetType type) {
-    Set<MetadataEntry> results = new HashSet<>();
+    List<MetadataEntry> results = new ArrayList<>();
```

Comment:
This was set so that if search for prefix based search if an Entity has more than one matching index we don't return that entity twice. I see why it needs to be list if we want to do scoring. 
But then shouldn't we perform search for only unique words in search query. Currently its an array. For example, consider this scenario: 
An user tagged two entities like this:
E1: Tags = tag1, tag 2
E2: Tags = tag3, tag3, tag3, tag3, tag3 (but since we only store unique tags we just have one tag for this)

Now search query is "tag1 tag2 tag3 tag3 tag3 tag3 tag3 tag3"
The current implementation will perform search for tag3 6 times and also will score E2 higher even though E1 matches two unique words in search query. Is this the behavior we want ? I think E1 should be ranked higher since it matched  two unique words.